packages/react-router/tests/router.test.ts

```diff
@@ -301,4 +301,45 @@ describe('encoding: splat param for /$', () => {
       'framework/react/guide/file-based-routing tanstack',
     )
   })
+
+  it('router.navigate with `params` and `to`, should transition from "/posts/tanner" to "/posts/tkdodo"', async () => {
+    const { router, routes } = createTestRouter(
+      createMemoryHistory({ initialEntries: ['/posts/tanner'] }),
+    )
+
+    await router.load()
+
+    expect(router.state.location.pathname).toBe('/posts/tanner')
+
+    await router.navigate({
+      to: routes.postIdRoute.to,
+      params: (prev: any) => {
+        console.log(prev)
+        return { ...prev, slug: 'tkdodo' }
+      }
+    })
+    // Not sure why I have to do this
+    await router.invalidate()
+
+    expect(router.state.location.pathname).toBe('/posts/tkdodo')
+  })
+
+  it('router.navigate with `params` and no `to`, should transition from "/posts/tanner" to "/posts/tkdodo"', async () => {
```

Comment:
With the tests in a new file, can your tests be expanded to include both a single path param (like you have now: `/posts/$slug`), and a more complicated setup like `/p/$project/$version/$framework` (URL like: `/p/router/latest/react`) and see if you can change only the $framework to "vue".